### PR TITLE
rename local instance variable for thread list

### PR DESCRIPTION
**Pull request by jmartin-tech**
Rename thread tracking instance variable. While there is no direct reproduction against a standalone console an error can appear when executing a scanner module for large sets of hosts. The following stack trace was observed.

```
[-] [2021.11.15-14:57:27] Auxiliary failed: TypeError nil can't be coerced into Integer
[-] [2021.11.15-14:57:27] Call stack:
[-] [2021.11.15-14:57:27]   /metasploit-framework/lib/msf/core/auxiliary/scanner.rb:255:in `+'
[-] [2021.11.15-14:57:27]   /metasploit/metasploit-framework/lib/msf/core/auxiliary/scanner.rb:255:in `block in run'
[-] [2021.11.15-14:57:27]   /metasploit/metasploit-framework/lib/msf/core/auxiliary/scanner.rb:255:in `map'
[-] [2021.11.15-14:57:27]   /metasploit/metasploit-framework/lib/msf/core/auxiliary/scanner.rb:255:in `run'
```

Also updates batch scanner modules to no longer crash when being able to unable to calculate a scanner thread's batch size

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] `use <SOME RANDOM AUX SCANNER MODULE>`
- [ ] **Verify** at least 3 to 5 scanner modules function as expected.
- [ ] Internal R7 team should also validate Pro discovery for large host ranges stabilizes.

### CODE REVIEW

File: lib/msf/core/auxiliary/scanner.rb [hunk 1]
```diff
@@ -63,7 +63,7 @@ def run
   @range_percent = 0
 
   threads_max = datastore['THREADS'].to_i
-  @tl = []
+  @stl = []
```
Comment (adfoster-r7) [inline lib/msf/core/auxiliary/scanner.rb:66]: What are your thoughts on renaming this variable to be more descriptive

```suggestion
  @thread_list = []
```

Comment (jmartin-r7) [reply lib/msf/core/auxiliary/scanner.rb]: Works for me, updated!

File: lib/msf/core/auxiliary/scanner.rb [hunk 2]
```diff
@@ -240,19 +240,19 @@ def run
...
       end
 
       # Attempt to wait for the oldest thread for a second,
       # remove any finished threads from the list
       # and continue on.
       tla = 0
-      @tl.map {|t| tla += t[:batch_size] }
-      @tl.first.join(1)
-      @tl.delete_if { |t| not t.alive? }
+      @thread_list.map {|t| tla += t[:batch_size] if t[:batch_size] }
```
Comment (adfoster-r7) [inline lib/msf/core/auxiliary/scanner.rb:251]: If we land this PR, are we okay with the potential of skipping over hosts unintentionally?

i.e. Before this change, a crash was observed - and it was clear that something had gone wrong, and the scan needed to be restarted to verify all of the targets.

After this change, we handle this unknown scenario more gracefully, but potentially there was still issues that stopped hosts from being scanned, and they're not longer surfaced to the user

Comment (jmartin-r7) [reply lib/msf/core/auxiliary/scanner.rb:251]: This should not impact the execution, this guard will only impact the calculations to skip `counting` objects in a thread that is not reporting a `:batch_size`.

Execution is based on `@thread_list` containing any threads that are still `.alive?`.

Comment (adfoster-r7) [reply lib/msf/core/auxiliary/scanner.rb:251]: I was less worried about the overall host count tracking - and was more worried about why a thread went awol and dropped local storage values, as there's the potential that it's just skipped hosts.

Happy for this to land in the interim though

**Comment by adfoster-r7** [comment]
## Release Notes

Updates batch scanner modules to no longer crash when being able to unable to correctly calculate a scanner thread's batch size

**Pull request merged**